cmd/lakectl/cmd/repo.go

```diff
@@ -93,10 +94,11 @@ var repoCreateCmd = &cobra.Command{
 			DefaultBranch:    defaultBranch,
 			ID:               &u.Repository,
 		})
-		if err != nil {
+		if err != nil && strings.Contains(err.Error(), "duplicate") {
+			DieFmt("Repository '%s' already exists.\n", u.Repository)
```

Comment:
Thanks!  This unfortunately ties the client to the server implementation; that would be unfortunate -- e.g. if the server implementation changed, or even if the PostgreSQL error message changed.

Also, doing it here on this client means that any other program that uses the [lakeFS API](https://docs.lakefs.io/reference/api.html) will not benefit from this change.

I would prefer that you detect this type of error on the server side.  E.g. the create-repo operation inside `catalog/` should identify this duplicate key condition (because -- unlike the client -- it is necessarily acquainted with the database), and report a new error type.  The client probably does not even need to be changed.

Reply:
@arielshaqed, thanks that makes sense! I will make those changes